File: capacity-club-api/src/domain-modules/training-circuit/model/entity/training-circuit.entity.ts
import { BaseEntity } from '@common/model';
import { UniqueId, uniqueIdTransformer } from '@common/model/unique-id';
import { Transform } from 'class-transformer';
import { ExerciseTraining } from 'domain-modules/exercise-training/model/entity/exercise-training.entity';
import { Workout } from 'domain-modules/workout/model/entity/workout.entity';

import {
  Entity,
  PrimaryColumn,
  Column,
  OneToMany,
  ManyToOne,
  JoinColumn,
} from 'typeorm';

/**
 * Entity representing a training circuit.
 * A training circuit is a collection of exercise training and is part of a workout.
 */
@Entity()
export class TrainingCircuit extends BaseEntity {
  /**
   * Unique identifier for the training circuit.
   * Uses a custom transformer for serialization.
   */
  @PrimaryColumn('varchar')
  @Transform(uniqueIdTransformer.to, { toClassOnly: true })
  @Transform(uniqueIdTransformer.from, { toPlainOnly: true })
  training_circuit_id: UniqueId;
  @Column({ nullable: false })
  title: string;

  @OneToMany(
    () => ExerciseTraining,
    (exerciseTraining) => exerciseTraining.training_circuit,
    {
      cascade: true,
      eager: true,
    },
  )
  exercise_training_list: ExerciseTraining[];

  @ManyToOne(() => Workout, (workout) => workout.training_circuits)
  @JoinColumn({ referencedColumnName: 'workout_id', name: 'workout_id' })
  workout: Workout;
}

File: capacity_club_mobile_app/lib/features/core/module/training-circuit/domain/entity/training-circuit.entity.dart
import 'package:capacity_club_mobile_app/features/core/module/exercise-training/domain/entity/exercise-training.entity.dart';

class TrainingCircuit {
  final String training_circuit_id;
  final String title;
  final List<ExerciseTraining> exercise_training_list;

  TrainingCircuit(
      {required this.training_circuit_id,
      required this.title,
      required this.exercise_training_list});

  factory TrainingCircuit.empty() {
    return TrainingCircuit(
        training_circuit_id: '', title: '', exercise_training_list: []);
  }
}
